nixos/modules/services/networking/zerotierone.nix

```diff
@@ -83,6 +97,16 @@ in
           ln -s ${localConfFile} ${localConfFilePath}
         '';
 
+      postStart = lib.optionalString cfg.generateSystemdNetworkdConfig ''
+        # zerotier-systemd-manager requires http://localhost:9993 to be available,
+        # which may take some short time.
+
+        sleep 5
```

Comment:
A better approach would be to test for the port. In bash you can use the magic /dev/tcp namespace for that. Eg:

```sh
count=0
while [[ $count -lt 5 ]]; do
  if exec 5<>/dev/tcp/localhost/9993; then
    break
  fi
  sleep 1
  $((count++))
done
```

Reply:
I have tried this approach, however it proved unreliable, with the connection getting stuck about 1/10 times

Reply:
There is also a time between the port being open, and the HTTP request succeeding (at least I seem to have encountered it at least once). I wrote a bash script to make the http request and ran it in `bash -c` with timeout, however it was too complicated/magic-like to publish.